chromadb/test/property/test_clean_log.py

```diff
@@ -0,0 +1,89 @@
+from typing import Generator, cast
+from overrides import overrides
+import pytest
+from chromadb.api import ServerAPI
+from chromadb.config import System
+from chromadb.db.base import get_sql
+from chromadb.db.impl.sqlite import SqliteDB
+from chromadb.ingest import Producer
+from pypika import Table, functions
+from hypothesis.stateful import (
+    run_state_machine_as_test,
+    invariant,
+)
+
+from chromadb.test.conftest import sqlite_fixture, sqlite_persistent_fixture
+from chromadb.test.property.test_embeddings import (
+    EmbeddingStateMachineBase,
+    EmbeddingStateMachineStates,
+)
+
+
+def count_embedding_queue_rows(sqlite: SqliteDB) -> int:
+    t = Table("embeddings_queue")
+    q = sqlite.querybuilder().from_(t).select(functions.Count(t.seq_id))
+
+    with sqlite.tx() as cur:
+        sql, params = get_sql(q, sqlite.parameter_format())
+        result = cur.execute(sql, params)
+        return cast(int, result.fetchone()[0])
+
+
+class LogCleanEmbeddingStateMachine(EmbeddingStateMachineBase):
+    has_collection_mutated = False
+    system: System
+
+    def __init__(self, system: System) -> None:
+        api = system.instance(ServerAPI)
+        self.system = system
+        super().__init__(api)
+
+    @invariant()
+    def log_empty_after_cleaning(self) -> None:
+        producer = self.system.instance(Producer)
+        sqlite = self.system.instance(SqliteDB)
+
+        producer.clean_log(self.collection.id)
+        num_rows = count_embedding_queue_rows(sqlite)
+
+        if self.has_collection_mutated:
+            # Must always keep one entry to avoid reusing seq_ids
+            assert num_rows == 1
+        else:
+            assert num_rows == 0
+
+    @overrides
+    def on_state_change(self, new_state: str) -> None:
+        if new_state != EmbeddingStateMachineStates.initialize:
+            self.has_collection_mutated = True
+
+
+@pytest.fixture(params=[sqlite_fixture, sqlite_persistent_fixture])
+def any_sqlite(request: pytest.FixtureRequest) -> Generator[System, None, None]:
+    yield from request.param()
```

Comment:
we can't do `system` here instead because that fixture gives us the `fastapi` system instance, which has the server in a separate process
we need the server in the same process so we can call `.clean_log()`